pkg/controller/gitopsconfig/controller.go

```diff
@@ -207,31 +208,51 @@ func ContainsTrigger(instance *gitopsv1alpha1.GitOpsConfig, triggeType string) b
 }
 
 // createJob creates a new gitops job for the passed instance
-func (r *Reconciler) createJob(jobtype string, instance *gitopsv1alpha1.GitOpsConfig) error {
+func (r *Reconciler) createJob(jobtype string, instance *gitopsv1alpha1.GitOpsConfig) (reconcile.Result, error) {
 	//TODO add logic to ignore if another job was created sooner than x (5 minutes?) time and it is still running.
 	mergedata := util.JobMergeData{
 		Config: *instance,
 		Action: jobtype,
 	}
+	// looking up for running jobs
+	jobList := &batchv1.JobList{}
+	err = r.client.List(context.TODO(), &client.ListOptions{
+		Namespace: instance.Namespace,
+	}, jobList)
+
+	if err != nil {
+		log.Error(err, "unable to list the jobs")
+		return reconcile.Result{}, err
+	}
+
+	for _, jobFromList := range jobList.Items {
+		if isOwner(instance, &jobFromList.ObjectMeta) {
+			if jobFromList.Status.Active != 0 {
```

Comment:
Please merge this `if` condition with the one above, like this:

    if isOwner(...) && jobFromList.Status.Active != 0 {